forgerock-core/src/main/java/org/forgerock/android/auth/EncryptedSharedPreferences.kt

```diff
@@ -0,0 +1,29 @@
+package org.forgerock.android.auth
+
+import android.content.Context
+import android.content.SharedPreferences
+import android.os.Build
+import androidx.annotation.RequiresApi
+import androidx.security.crypto.EncryptedSharedPreferences
+import androidx.security.crypto.MasterKeys
+
+class EncryptedPreferences {
+    companion object {
+        fun getInstance(
+            context: Context,
+            masterKey: String = "forgerock_master"
```

Comment:
Since the SDK is opensource, everyone will have the masterkey, is EncryptedSharedPreferences secure? If someone has the SharePreferences and the masterkey, can they decrypt the data? Where should we store the masterkey?

Reply:
@witrisna i changed this to a filename, the file where we store all the encryped data, i dont think they can decrypt the data .

Reply:
MasterKeys class allows you to get a recommended default master key with default settings. This is how you can get the default master key:

val keyAlias = MasterKeys.getOrCreate(MasterKeys.AES256_GCM_SPEC)

MasterKeys class contains methods to create and obtain master keys in Android Keystore. getOrCreate method gets or creates the master key provided. The encryption scheme is a required field, and for now, there is only one value that you can use, MasterKeys.AES256_GCM_SPEC.